### PR TITLE
Feature/technical expand firestore adapters

### PR DESCRIPTION
**Pull request by santidrj**

### CODE REVIEW

File: src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollectionAdapter.java [hunk 1]
```diff
@@ -51,6 +60,20 @@ public DocumentReference getCollectionParent(@NonNull String path) {
...
+    @NonNull
+    @Override
+    public List<DocumentSnapshot> listAllCollectionDocumentSnapshots(@NonNull String path)
+        throws DatabaseAccessException {
+        Iterable<DocumentReference> iterable = listAllCollectionDocuments(path);
+        List<DocumentSnapshot> snapshots = new ArrayList<>();
+        for (DocumentReference doc : iterable) {
+            try {
+                snapshots.add(documentAdapter.getDocumentSnapshot(doc.getPath()));
+            } catch (DocumentException ignore) { }
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreCollectionAdapter.java:72]: Codacy Issue found: [Empty catch block.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5465948)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Coverage increased per file
===========================
+ src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreDocumentAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5465948&bid=17915413)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ src/main/java/org/pesmypetcare/webservice/firebaseservice/adapters/firestore/FirestoreDocumentAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5465948&bid=17915413)

File: src/test/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreDocumentAdapterTest.java [hunk 2]
```diff
@@ -262,39 +381,35 @@ public void getDocumentField() throws DatabaseAccessException, DocumentException
                 willReturn(aString).given(documentSnapshot).get(anyString());
 
                 Object result = adapter.getDocumentField(documentPath, field);
-                assertEquals(aString, result,
-                    "Should return " + aString + " as the value of the " + field + " in the " + "document located at "
-                        + documentPath);
+                assertEquals(aString, result, "Should return " + aString + " as the value of the " + field + " in the "
+                    + "document located at " + documentPath);
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreDocumentAdapterTest.java:385]: Codacy Issue found: [The String "document located at " appears 8 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5465948)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5465948&bid=17915413)

File: src/main/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreCollectionAdapter.java [hunk 3]
```diff
@@ -51,6 +60,20 @@ public DocumentReference getCollectionParent(@NonNull String path) {
...
+    @NonNull
+    @Override
+    public List<DocumentSnapshot> listAllCollectionDocumentSnapshots(@NonNull String path)
+        throws DatabaseAccessException {
+        Iterable<DocumentReference> iterable = listAllCollectionDocuments(path);
+        List<DocumentSnapshot> snapshots = new ArrayList<>();
+        for (DocumentReference doc : iterable) {
+            try {
+                snapshots.add(documentAdapter.getDocumentSnapshot(doc.getPath()));
+            } catch (DocumentException ignore) { }
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreCollectionAdapter.java:72]: Codacy Issue found: [Empty catch block.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5465948)

File: src/test/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreDocumentAdapterTest.java [hunk 4]
```diff
@@ -97,9 +97,22 @@ public void createDocument() {
         given(collectionReference.document()).willReturn(documentReference);
         given(batch.create(any(DocumentReference.class), anyMap())).willReturn(batch);
 
-        adapter.createDocument(collectionPath, fields, batch);
+        DocumentReference ref = adapter.createDocument(collectionPath, fields, batch);
         verify(collectionReference).document();
         verify(batch).create(same(documentReference), anyMap());
+        assertEquals(documentReference, ref, "Should return the document reference created");
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreDocumentAdapterTest.java:103]: Codacy Issue found: [The String "Should return the document reference created" appears 7 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5465948)

File: src/test/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreCollectionAdapterTest.java [hunk 5]
```diff
@@ -113,34 +139,35 @@ public void getDocumentsWhereFieldEqualToValue() {
...
 
         @Test
         public void getDocumentsWhereArrayContainsValue() {
             given(collectionReference.whereArrayContains(same(array), same(value))).willReturn(query);
             Query result = collectionAdapter.getDocumentsWhereArrayContains(collectionPath, array, value);
             assertEquals(query, result,
-                "Should return all documents from " + collectionPath + " where the array " + array + " contains "
-                    + value);
+                         "Should return all documents from " + collectionPath + " where the array " + array
+                             + " contains " + value);
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreCollectionAdapterTest.java:161]: Codacy Issue found: [The String " contains " appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5465948)

File: src/test/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreCollectionAdapterTest.java [hunk 6]
```diff
@@ -113,34 +139,35 @@ public void getDocumentsWhereFieldEqualToValue() {
             given(collectionReference.whereEqualTo(same(field), same(value))).willReturn(query);
             Query result = collectionAdapter.getDocumentsWhereEqualTo(collectionPath, field, value);
             assertEquals(query, result,
-                "Should return all documents from " + collectionPath + " where " + field + " is equals to " + value);
+                         "Should return all documents from " + collectionPath + " where " + field + " is equals to "
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreCollectionAdapterTest.java:142]: Codacy Issue found: [The String " is equals to " appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5465948)

File: src/test/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreDocumentAdapterTest.java [hunk 7]
```diff
@@ -262,39 +381,35 @@ public void getDocumentField() throws DatabaseAccessException, DocumentException
                 willReturn(aString).given(documentSnapshot).get(anyString());
 
                 Object result = adapter.getDocumentField(documentPath, field);
-                assertEquals(aString, result,
-                    "Should return " + aString + " as the value of the " + field + " in the " + "document located at "
-                        + documentPath);
+                assertEquals(aString, result, "Should return " + aString + " as the value of the " + field + " in the "
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreDocumentAdapterTest.java:384]: Codacy Issue found: [The String " in the " appears 8 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5465948)

File: src/test/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreCollectionAdapterTest.java [hunk 8]
```diff
@@ -163,17 +190,17 @@ public void getCollectionGroupDocumentsWhereFieldEqualToValue() {
             given(query.whereEqualTo(same(field), same(value))).willReturn(query);
             Query result = collectionAdapter.getCollectionGroupDocumentsWhereEqualTo(collectionGroup, field, value);
             assertEquals(query, result,
-                "Should return all documents from the collections whose id is " + collectionGroup + " where " + field
-                    + " is equals to " + value);
+                         "Should return all documents from the collections whose id is " + collectionGroup + " where "
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreCollectionAdapterTest.java:193]: Codacy Issue found: [The String "Should return all documents from the collections whose id is " appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5465948)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Coverage increased per file
===========================
+ src/main/java/org/pesmypetcare/webservice/thirdpartyservices/adapters/firestore/FirestoreDocumentAdapter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5465948&bid=17915413)

**Review by AlbertPG — APPROVED** [review]
Okey, I see no problem

**Review by marclll — APPROVED** [review]
OK

**Pull request merged**